src/workerd/api/gpu/gpu-buffer.c++

```diff
@@ -46,18 +56,142 @@ GPUBuffer::getMappedRange(jsg::Optional<GPUSize64> offset,
 
   JSG_REQUIRE(ptr, TypeError, "could not obtain mapped range");
 
-  auto arrayBuffer = kj::arrayPtr((byte *)ptr, s);
-  mapped_.add(Mapping{start, end, arrayBuffer});
+  struct Context {
+    jsg::Ref<GPUBuffer> buffer;
+  };
+  auto ref = JSG_THIS;
+  // We're creating this context object in order for it to be available when the
+  // callback is invoked. It owns a persistent reference to this GPUBuffer
+  // object to ensure that it still lives while the arraybuffer is in scope.
+  // This object will be deallocated when the callback finishes.
+  auto ctx = new Context{ref.addRef()};
+  std::shared_ptr<v8::BackingStore> backing = v8::ArrayBuffer::NewBackingStore(
+      ptr, s,
+      [](void *data, size_t length, void *deleter_data) {
+        // we have a persistent reference to GPUBuffer so that it lives at least
+        // as long as this arraybuffer
+        // Note: this is invoked outside the JS isolate lock
+        auto c = std::unique_ptr<Context>(static_cast<Context *>(deleter_data));
+      },
+      ctx);
+
+  v8::Local<v8::ArrayBuffer> arrayBuffer =
+      v8::ArrayBuffer::New(js.v8Isolate, backing);
+  arrayBuffer->SetDetachKey(
+      V8AtomicString(js.v8Isolate, kWebGPUBufferMappingDetachKey));
```

Comment:
The detach key here can really be any opaque JS value, it doesn't need to be a string. It also doesn't need to be a unique value for every call. I would recommend just creating a new object (e.g. `v8::Object::New(...)` when the `GPUBuffer` instance is created and use it as the detach key for every instance.

Reply:
Done.